crates/router/src/core/payments.rs

```diff
@@ -760,47 +743,151 @@ pub async fn add_process_sync_task(
     Ok(())
 }
 
-pub async fn route_connector<F>(
+pub fn update_straight_through_routing<F>(
+    payment_data: &mut PaymentData<F>,
+    request_straight_through: serde_json::Value,
+) -> RouterResult<()>
+where
+    F: Send + Clone,
+{
+    let mut routing_data: storage::RoutingData = payment_data
```

Comment:
All places the error is lifted to `InternalServerError`.

We can implement like common lifting as follows
```rust
|| -> _ {
   \\ code block with common error with attach_printable if needed
}().change_context(errors::ApiErrorResponse::InternalServerError)

```

Reply:
we can use this approach in other places too if all errors inside the scope are common.

Reply:
In that case we cannot attach specific printables on the `InternalServerError` context. Altho, after taking a look, I think it's okay to attach all printables on the `ParsingError` context in this case. So i guess we can go ahead with this. I'll make the change.

Reply:
Since all errors are the same here, just lifted the ApiErrorResponse change_context outside the function.